x-pack/plugins/task_manager/server/task_store.ts

```diff
@@ -278,6 +279,12 @@ export class TaskStore {
       )
     );
 
+    // the sort should use score first, but only if there are pinned tasks
```

Comment:
Can we add a comment about score being lower for old tasks and that that is the opposite to what TM actually needs, so whenever score is used TM is likely to pick up newer expired tasks rather than the older ones.

Reply:
I'll add more explanation - but it's not entirely clear to me that the score IS strictly lower for older tasks - I didn't actually go down the rabbit hole to figure out how the scores were calculated :-).  That would be my guess as well - that older docs score lower - but sometimes things actually worked out fairly well, so I think there may be more to it than just that.  I've been assuming the score is basically a random number at this point, except for the pinned tasks that *will* score better ...